db/collection_test.go

```diff
@@ -102,6 +102,48 @@ func TestNewCollection(t *testing.T) {
 			t.Fatalf("expected %d indexes, got %d", 2, len(indexes))
 		}
 	})
+	t.Run("WithWriteValidator", func(t *testing.T) {
+		t.Parallel()
+		db, clean := createTestDB(t)
+		defer clean()
+		c, err := db.NewCollection(CollectionConfig{
+			Name:           "Dog",
+			Schema:         util.SchemaFromInstance(&Dog{}, false),
+			WriteValidator: `return true`,
+		})
+		checkErr(t, err)
+		dog := Dog{Name: "Fido", Comments: []Comment{}}
+		id, err := c.Create(util.JSONFromInstance(dog))
+		checkErr(t, err)
+		dog.ID = id
+		dog.Name = "Clyde"
+		err = c.Save(util.JSONFromInstance(dog))
+		checkErr(t, err)
+	})
+	t.Run("WithReadFilter", func(t *testing.T) {
+		t.Parallel()
+		db, clean := createTestDB(t)
+		defer clean()
+		c, err := db.NewCollection(CollectionConfig{
+			Name:   "Dog",
+			Schema: util.SchemaFromInstance(&Dog{}, false),
+			ReadFilter: `
```

Comment:
Here's an example of using a read filter to transform the instance. Buckets uses this to filter the paths shown in the bucket based on the bucket ACL.

Reply:
Oh, it can also mutate the instance. Scary but powerful :P
Now feel `ReadFilter` might be a bit short in semantics since isn't only deciding to "show/not_show" but possibly changing. I don't have a good candidate name though.

Reply:
Yep haha... I had a similar thought. Would `ReadTransform` or `ReadTransformer` be better?

Reply:
Uhm, yeah, but not sure how to conciliate both since "Transformer" doesn't sound like it could filter.
`ReadFilterTransformer` might be too much?